packages/kbn-plugin-generator/sao_template/template/public/app.js

```diff
@@ -1,34 +1,35 @@
-import moment from 'moment';
-import { uiModules } from 'ui/modules';
-import uiRoutes from 'ui/routes';
+import React from "react";
+import moment from "moment";
+import { uiModules } from "ui/modules";
+import chrome from "ui/chrome";
+import uiRoutes from "ui/routes";
+import { render, unmountComponentAtNode } from "react-dom";
 
-import 'ui/autoload/styles';
-import './less/main.less';
-import template from './templates/index.html';
+import "ui/autoload/styles";
+import "./less/main.less";
+import template from "./templates/index.html";
+import { Main } from "./components/main";
 
-uiRoutes.enable();
-uiRoutes
-  .when('/', {
-    template,
-    resolve: {
-      currentTime($http) {
-        return $http.get('../api/<%= name %>/example').then(function (resp) {
-          return resp.data.time;
-        });
-      }
-    }
+const app = uiModules.get("apps/<%= camelCase(name) %>");
+
+app.config($locationProvider => {
+  $locationProvider.html5Mode({
+    enabled: false,
+    requireBase: false,
+    rewriteLinks: false,
   });
+});
+
+function RootController($scope, $element, $http) {
+  const domNode = $element[0];
 
-uiModules
-  .get('app/<%= name %>', [])
-  .controller('<%= camelCase(name) %>HelloWorld', function ($scope, $route, $interval) {
-    $scope.title = '<%= startCase(name) %>';
-    $scope.description = '<%= description %>';
+  // render react to DOM
+  render(<Main title={"<%= name %>"} httpClient={$http}/>, domNode);
```

Comment:
nit: Can't the title arg just be `title="<%= name %>"`? It's just a string, there's no need to have React evaluate the value, is there?